bottom-bar/src/main/java/com/roughike/bottombar/TabParser.java

```diff
@@ -3,152 +3,192 @@
+import android.support.annotation.StringDef;
 import android.support.annotation.XmlRes;
 import android.support.v4.content.ContextCompat;
 
 import org.xmlpull.v1.XmlPullParserException;
 
 import java.io.IOException;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
 import java.util.ArrayList;
 import java.util.List;
 
+import static com.roughike.bottombar.TabParser.TabAttribute.ACTIVE_COLOR;
+import static com.roughike.bottombar.TabParser.TabAttribute.BADGE_BACKGROUND_COLOR;
+import static com.roughike.bottombar.TabParser.TabAttribute.BADGE_HIDES_WHEN_ACTIVE;
+import static com.roughike.bottombar.TabParser.TabAttribute.BAR_COLOR_WHEN_SELECTED;
+import static com.roughike.bottombar.TabParser.TabAttribute.ICON;
+import static com.roughike.bottombar.TabParser.TabAttribute.ID;
+import static com.roughike.bottombar.TabParser.TabAttribute.INACTIVE_COLOR;
+import static com.roughike.bottombar.TabParser.TabAttribute.TITLE;
+
 /**
  * Created by iiro on 21.7.2016.
+ *
+ * BottomBar library for Android
+ * Copyright (c) 2016 Iiro Krankka (http://github.com/roughike).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
  */
 class TabParser {
+    private static final String TAB_TAG = "tab";
+    private static final int AVG_NUMBER_OF_TABS = 5;
+    private static final int COLOR_NOT_SET = -1;
+    private static final int RESOURCE_NOT_FOUND = 0;
+
+    @NonNull
     private final Context context;
+
+    @NonNull
     private final BottomBarTab.Config defaultTabConfig;
+
+    @NonNull
     private final XmlResourceParser parser;
 
-    private ArrayList<BottomBarTab> tabs;
-    private BottomBarTab workingTab;
+    @Nullable
+    private List<BottomBarTab> tabs = null;
 
-    TabParser(Context context, BottomBarTab.Config defaultTabConfig, @XmlRes int tabsXmlResId) {
+    TabParser(@NonNull Context context, @NonNull BottomBarTab.Config defaultTabConfig, @XmlRes int tabsXmlResId) {
         this.context = context;
         this.defaultTabConfig = defaultTabConfig;
-
-        parser = context.getResources().getXml(tabsXmlResId);
-        tabs = new ArrayList<>();
-
-        parse();
+        this.parser = context.getResources().getXml(tabsXmlResId);
     }
 
-    private void parse() {
-        try {
-            parser.next();
-            int eventType = parser.getEventType();
-
-            while (eventType != XmlResourceParser.END_DOCUMENT) {
-                if(eventType == XmlResourceParser.START_TAG) {
```

Comment:
this would had trigger didn't matter what type of tag was starting, because the tag was only checked at the end